tests/README.md

```diff
@@ -1,44 +1,65 @@
 # Chalk Testing
 
-This directory contains all the basic functionality tests for chalk.
-Tests are run via docker compose in a separate `tests` container which
-internally uses the `pytest` framework.
+This directory contains all the basic functionality tests for chalk. Tests are run via `make tests`, which uses `docker compose` to run the `tests` container which internally uses the `pytest` framework.
 
-## Requirements
+While `pytest` can be used to run the tests directly, it is recommended to use the makefile script instead, as that provides a more consistent developer experience.
+
+## Setup
+
+Before running tests, ensure that `docker` and `docker compose` are installed.
+
+### Tests Container
+
+Please ensure you have an up-to-date container with:
+
+```sh
+docker compose build tests
+```
+
+Note that building the `tests` container will set up the `pytest` framework needed to run the tests, but the container will NOT have a copy of the chalk binary to test against. This is discussed in the next section.
 
 ### Chalk Binary
 
-Tests assume that a chalk binary is present in the root directory of
-the repo, and will use that binary in all tests cases.
+Upon starting a test run, the script will first look for an up-to-date chalk binary in the root directory of the repo. If there is no chalk binary, or if the chalk binary is not up to date (ie, there are local code changes that have not yet been built), the script will rebuild the chalk binary.
 
-The quickest way to get a binary with the current local changes is:
+WARNING: If there is already a chalk binary in the root directory that the script considers "out of date", this binary will be DELETED. If you want to keep this binary, move or rename it.
 
-1. Build chalk deps container:
+Tests will always rebuild the binary if it detects a code change in the underlying nim code. However, you will need to build `chalk` manually when switching between `release` and `debug` builds:
 
-   ```sh
-   docker compose build chalk
-   ```
+1. By default, the testing script will build a `release` binary for chalk, so if you want to test against the `debug` build instead you must build it yourself.
+2. If you have been working with a `debug` build so far, but want to switch to `release`, you will need to rebuild manually as the test script will not rebuild if there have not been any code changes.
 
-   (this step can be skipped if the `chalk` container is up to date)
+The quickest way to manually build a binary with the current local changes is:
 
-1. Compile `chalk`:
+1. Build chalk deps container:
 
-   ```sh
-   # root of the repo
-   make chalk
-   ```
+```sh
+docker compose build chalk
+```
 
-The second command should drop a `chalk` binary that is usable by tests.
+(this step can be skipped if the `chalk` container is up to date)
```

Comment:
same for rest of the text block inside lists. if you indent it, they will show up as being inside list. for example:

```
1. foo
   ```sh
   nested code
   ``
    description
2. bar
```

will render as:

1. foo
   ```sh
   nested code
   ```
    description
2. bar